src/db/group.rs

```diff
@@ -148,6 +157,156 @@ impl Group {
     pub fn get_expiry_time(&self) -> Option<&chrono::NaiveDateTime> {
         self.times.get_expiry()
     }
+
+    pub fn entries(&self) -> Vec<&Entry> {
+        let mut response: Vec<&Entry> = vec![];
+        for node in &self.children {
+            if let Node::Entry(e) = node {
+                response.push(e)
+            }
+        }
+        response
+    }
+
+    pub fn entries_mut(&mut self) -> Vec<&mut Entry> {
+        let mut response: Vec<&mut Entry> = vec![];
+        for node in &mut self.children {
+            if let Node::Entry(e) = node {
+                response.push(e)
+            }
+        }
+        response
+    }
+
+    pub fn groups_mut(&mut self) -> Vec<&mut Group> {
+        let mut response: Vec<&mut Group> = vec![];
+        for node in &mut self.children {
+            if let Node::Group(g) = node {
+                response.push(g);
+            }
+        }
+        response
+    }
+
+    fn replace_entry(&mut self, entry: &Entry) {
+        for node in &mut self.children {
+            match node {
+                Node::Group(g) => {
+                    g.replace_entry(entry);
+                }
+                Node::Entry(e) => {
+                    if e.uuid == entry.uuid {
+                        *e = entry.clone();
+                    }
+                }
+            }
+        }
+    }
+
+    pub fn find_entry_by_uuid(&self, id: Uuid) -> Option<&Entry> {
+        for node in &self.children {
```

Comment:
```suggestion
        self.children.iter_mut().for_each(|node| {
```

Reply:
Not sure what's the advantage of this proposition. We don't need a mutable reference to the node, and the proposed syntax is more verbose.

Reply:
The mut was a bit of an oversight. iter() should be sufficient. Overall, the iterator syntax should be preferred. It's cleaner and often faster than the for-loop.

Reply:
@phoerious 

> often faster than the for-loop.

Do you have a source for that? I would like to know why that's the case.

> It's cleaner 

I personally prefer the for syntax, especially in Rust. Notice the absence of parentheses on the line where the `for` is declared :sparkles: